readthedocs/proxito/middleware.py

```diff
@@ -382,7 +383,7 @@ def add_resolver_headers(self, request, response):
                     mime_encode=True,
                 )
 
-                response["X-RTD-Resolver-Filename"] = header_value
+                response["X-RTD-Resolver-Filename"] = iri_to_uri(header_value)
```

Comment:
`iri_to_uri` should be applied before calling `escape()`. We should never change values that are already escaped.